fire/SFMainMod.F90

```diff
@@ -916,28 +926,52 @@ subroutine  crown_damage ( currentSite )
+                crown_depth = currentCohort%hite*EDPftvarcon_inst%crown(currentCohort%pft) 
+                height_cbb  = currentCohort%hite - crown_depth
+
+                
+                ! Evaluate for passive crown fire ignition
+                if (EDPftvarcon_inst%crown_fire(currentCohort%pft) == 1) then
+                   
+                   ! Crown fuel ignition potential, EQ 8 Bessie and Johnson 1995
+                   currentCohort%passive_crown_FI = (0.01_r8 * height_cbb * SF_val_crown_ignition_energy)**1.5_r8
+
+                   if (currentCohort%passive_crown_FI >= crown_fire_threshold) then ! 200 kW/m = threshold for crown fire potential
+                      
+                      ! Initiation of passive crown fire, EQ 14 Bessie and Johnson 1995
+                      currentCohort%ignite_crown = currentPatch%FI/currentCohort%passive_crown_FI
+                      
+                      if (currentCohort%ignite_crown >= 1.0_r8) then
+                         currentCohort%crown_fire_flg = 1  ! passive crown fire ignited
+                         currentCohort%fraction_crown_burned =  1.0_r8
+                      ! else ! evaluate crown damage based on scorch height
+                      endif ! ignite passive crown fire
+                   ! else no crown fire today
+                   endif ! crown fire intensity
+                ! else ! not crown fire plant
+                endif ! evaluate passive crown fire
+                
+                ! Flames lower than bottom of canopy.
+                ! height_cbb is clear branch bole height or height of bottom of canopy 
+                if (currentPatch%SH <= height_cbb) .and. (currentCohort%crown_fire_flg == 0) then &
                    currentCohort%fraction_crown_burned = 0.0_r8
-                else
-                   ! Flames part of way up canopy. 
-                   ! Equation 17 in Thonicke et al. 2010. 
-                   ! flames over bottom of canopy but not over top.
-                   if ((currentCohort%hite > 0.0_r8).and.(currentPatch%SH >=  &
-                        (currentCohort%hite-currentCohort%hite*EDPftvarcon_inst%crown(currentCohort%pft)))) then 
-
-                           currentCohort%fraction_crown_burned =  (currentPatch%SH-currentCohort%hite*(1.0_r8- &
-                                EDPftvarcon_inst%crown(currentCohort%pft)))/(currentCohort%hite* &
-                                EDPftvarcon_inst%crown(currentCohort%pft)) 
-
-                   else 
-                      ! Flames over top of canopy. 
-                      currentCohort%fraction_crown_burned =  1.0_r8 
-                   endif
 
-                endif
+                ! Flames part way into canopy
+                ! Equation 17 in Thonicke et al. 2010
+                elseif ((currentCohort%hite > 0.0_r8).and.(currentPatch%SH > height_cbb) &
+                     .and. (currentCohort%crown_fire_flg == 0) then  &
```

Comment:
Similar here. Remove the `&` at the end and add one more parenthesis at the end for balance:
```suggestion
                     .and. (currentCohort%crown_fire_flg == 0)) then
```